unlock-app/src/components/interface/pagination/PageNumbers.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { PageGroup, PageNumber, PageNumberActive } from './Pagination'
+import { PGN_MAX_NUMBER_OF_PAGES_TO_SHOW_ALL } from '../../../constants'
+
+const PageNumbers = ({ numberOfPages, currentPage, goToPage }) => {
+  const pageNumber = number => {
+    return number === currentPage ? (
+      <PageNumberActive
+        key={number.toString()}
+        onClick={() => goToPage(number)}
+      >
+        {number}
+      </PageNumberActive>
+    ) : (
+      <PageNumber key={number.toString()} onClick={() => goToPage(number)}>
+        {number}
+      </PageNumber>
+    )
+  }
+
+  // If number of pages less than 10, show all 10 page numbers
+  if (numberOfPages <= PGN_MAX_NUMBER_OF_PAGES_TO_SHOW_ALL) {
+    return (
+      <PageGroup>
+        {Array(numberOfPages)
+          .fill()
+          .map((_, pn) => {
+            return pageNumber(pn + 1)
+          })}
+      </PageGroup>
+    )
+  }
+
+  const HALF_MAX_PAGES_TO_SHOW_ALL = PGN_MAX_NUMBER_OF_PAGES_TO_SHOW_ALL / 2
+
+  // If number of pages greater than 10,  show first five and last five (1 2 3 4 5 .... 26 27 28 29 30)
+  return (
+    <PageGroup>
+      {//show first five page numbers
+        Array(HALF_MAX_PAGES_TO_SHOW_ALL)
+          .fill()
+          .map((_, pn) => {
+            return pageNumber(pn + 1)
+          })}
+      {//show first few dots
+        currentPage - 1 === HALF_MAX_PAGES_TO_SHOW_ALL ? (
```

Comment:
Could you extract this complex logic into some constants and then render them so it is easier to follow the render logic?